### PR TITLE
Disconnect from channels when going away, reconnect when back.

### PR DESCRIPTION
**Pull request by eatnumber1**
This is a rough patch which will part from channels when going away, and rejoin when you return. This is useful for me because I like to hook up CamperVan to my IRC bouncer and be able to simply `/away` once for all of the servers my bouncer connects to.

### CODE REVIEW

**Comment by zerowidth** [comment]
I'm not sure this would be expected behavior for most users, especially since you'd lose any history by leaving a room. Campfire does have a vague sense of "idle", as I recall, but nothing you can set programmatically. Maybe a command-line option, e.g. `--part-on-away` defaulting to false would be better here.

**Comment by eatnumber1** [comment]
Sounds reasonable. I've added an option.

File: lib/camper_van/ircd.rb [hunk 1]
```diff
@@ -41,10 +44,13 @@ class IRCD
     # Public: initialize an IRC server connection
     #
     # client - the EM connection representing the IRC client
-    def initialize(client)
+    def initialize(client, options = {})
       @client = client
       @active = true
       @channels = {}
+      @away = false
+      @saved_channels = nil
```
Comment (zerowidth) [inline lib/camper_van/ircd.rb]: I'm thinking set this to an empty array, so _just in case_ if there's an unexpected state change, line 255 won't blow up. Same for line 258.

**Comment by zerowidth** [comment]
Lookin' good so far. Mind adding a couple tests just to make sure the happy path is working?

**Comment by eatnumber1** [comment]
Done. I also made /away appear to work to the client even without `--part-on-away` which should prevent the state my client is in currently whereby it thinks it's away and can't get out of it.

The only test I wasn't exactly sure how to write was the "return from away when --part-on-away is set"

**Comment by zerowidth** [comment]
Merged, thanks! It'll be part of the next release, which is Coming Soon™

**Comment by eatnumber1** [comment]
Awesome, thanks!

**Pull request merged**